modulus/experimental/resdiff/generate.py

```diff
@@ -318,278 +315,154 @@ def load_pickle(network_pkl, rank):
-    help="Stochasticity strength",
-    metavar="FLOAT",
-    type=click.FloatRange(min=0),
-    default=0,
-    show_default=True,
-)
-@click.option(
-    "--S_min",
-    "S_min",
-    help="Stoch. min noise level",
-    metavar="FLOAT",
-    type=click.FloatRange(min=0),
-    default=0,
-    show_default=True,
-)
-@click.option(
-    "--S_max",
-    "S_max",
-    help="Stoch. max noise level",
-    metavar="FLOAT",
-    type=click.FloatRange(min=0),
-    default="inf",
-    show_default=True,
-)
-@click.option(
-    "--S_noise",
-    "S_noise",
-    help="Stoch. noise inflation",
-    metavar="FLOAT",
-    type=float,
-    default=1,
-    show_default=True,
-)
-@click.option(
-    "--solver",
-    help="Ablate ODE solver",
-    metavar="euler|heun",
-    type=click.Choice(["euler", "heun"]),
-)
-@click.option(
-    "--disc",
-    "discretization",
-    help="Ablate time step discretization {t_i}",
-    metavar="vp|ve|iddpm|edm",
-    type=click.Choice(["vp", "ve", "iddpm", "edm"]),
-)
-@click.option(
-    "--schedule",
-    help="Ablate noise schedule sigma(t)",
-    metavar="vp|ve|linear",
-    type=click.Choice(["vp", "ve", "linear"]),
-)
-@click.option(
-    "--scaling",
-    help="Ablate signal scaling s(t)",
-    metavar="vp|none",
-    type=click.Choice(["vp", "none"]),
-)
-@click.option(
-    "--data_type", help="String to include the data type", metavar="cwb|era5|cwb-era5"
-)
-@click.option(
-    "--data_config",
-    help="String to include the data config",
-    metavar="full_field_val_crop64",
-)
-@click.option(
-    "--task",
-    help="String to include the task",
-    metavar="sr|pred",
-    type=click.Choice(["sr", "pred"]),
-)
-@click.option(
-    "--sample_res",
-    help="String to include the task",
-    metavar="full|patch",
-    type=click.Choice(["full", "patch"]),
-)
-@click.option(
-    "--pretext",
-    help="String to include the task",
-    metavar="full|patch",
-    type=click.Choice(["gen", "reg", "res"]),
-)
-@click.option("--res_edm", help="if residual based edm used", is_flag=True)
-@click.option(
-    "--network_reg",
-    "network_reg_pkl",
-    help="Network pickle filename",
-    metavar="PATH|URL",
-    type=str,
-)
-@click.option(
-    "--sampling_method",
-    help="Sampling method for generation",
-    metavar="stochastic|deterministic|none",
-    type=click.Choice(["stochastic", "deterministic", "none"]),
-)
-
-# def main(data_config, task, data_type, det_batch=None, gen_batch=None):
-def main(max_times: Optional[int], seeds: List[int], **kwargs):
+@hydra.main(version_base="1.2", config_path="conf", config_name="config_generate")
+def main(cfg: DictConfig) -> None:
+    """Generate random images using the techniques described in the paper
+    "Elucidating the Design Space of Diffusion-Based Generative Models".
+    """
 
-    opts = dnnlib.EasyDict(kwargs)
+    # Parse options
+    res_ckpt_filename = getattr(cfg, "res_ckpt_filename")
+    reg_ckpt_filename = getattr(cfg, "reg_ckpt_filename")
```

Comment:
Since the checkpoint files are required shouldn't these be check rather than assign a default value?

Reply:
I'm not sure if I understand this correctly, but these are required args and don't have default values.